library/modules/Textures.cpp

```diff
@@ -118,42 +150,66 @@ std::vector<TexposHandle> slice_tileset(SDL_Surface* surface, int tile_px_w, int
     return handles;
 }
 
-TexposHandle Textures::loadTexture(SDL_Surface* surface) {
+TexposHandle Textures::loadTexture(SDL_Surface* surface, bool reserved) {
     if (!surface)
         return 0; // should be some error, i guess
+    if (loading_state) {
+        ERR(textures).printerr("unable to load texture during game loading\n");
+        return 0;
+    }
 
     auto handle = reinterpret_cast<uintptr_t>(surface);
     g_handle_to_surface.emplace(handle, surface);
     surface->refcount++; // prevent destruct on next FreeSurface by game
-    auto texpos = add_texture(surface);
-    g_handle_to_texpos.emplace(handle, texpos);
+    if (reserved) {
+        auto texpos = reserved_range.get_new_texpos();
+        if (texpos == -1) {
+            ERR(textures).printerr("reserved range limit has been reached, use dynamic range\n");
+            return 0;
```

Comment:
how about if this only becomes an error if `loading_state == true`? otherwise we can automatically fall back to making it a dynamic texture.

Reply:
it can be, but dev can count on reserved range during development (it is a special range  for special purposes)
what if user will install to many plugins and will getting error because some texposes will be dynamic?

Reply:
I think it's more important that this code always returns a valid handle. whether a texture is reserved or dynamic should not materially matter to the *code* that uses it. the experience will just gracefully degrade as pressure is increased, which is what we want. If a texture cannot be loaded because there is no room in the reserved buffer and loading_state is true, then we should return a valid handle that points to texpos (0) and we should try to add the real texture to the vector the next time the handle is resolved.

Reply:
in case of texpos (0) and  `loading_state`user will receive incorrect texpos for ever call..
next handle resolve can be in loading_state as well

Reply:
if you insist on loading during loading state, so it can be delayed queue, that will be loaded after loading state become false